src/consensus/blockruleindex.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_BLOCKRULEINDEX_H
+#define BITCOIN_BLOCKRULEINDEX_H
+
+#include "versionbits.h"
+
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <stdint.h>
+
+class CBlockIndex;
+
+namespace Consensus
+{
+
+namespace VersionBits
```

Comment:
Is it really worth it to have so many namespaces (specially when you are avoiding to use them explicitly in so many cases with ```using namespace```)? Isn't the Consensus namespace enough?